pixeldata/src/gdcm.rs

```diff
@@ -144,6 +145,164 @@ where
             window,
         })
     }
+
+    fn decode_pixel_data_frame(&self, frame: u32) -> Result<DecodedPixelData<'_>> {
+        use super::attribute::*;
+
+        let pixel_data = pixel_data(self).context(GetAttributeSnafu)?;
+
+        let cols = cols(self).context(GetAttributeSnafu)?;
+        let rows = rows(self).context(GetAttributeSnafu)?;
+
+        let photometric_interpretation =
+            photometric_interpretation(self).context(GetAttributeSnafu)?;
+        let pi_type = GDCMPhotometricInterpretation::from_str(photometric_interpretation.as_str())
+            .map_err(|_| {
+                UnsupportedPhotometricInterpretationSnafu {
+                    pi: photometric_interpretation.clone(),
+                }
+                .build()
+            })?;
+
+        let transfer_syntax = &self.meta().transfer_syntax;
+        let registry =
+            TransferSyntaxRegistry
+                .get(&&transfer_syntax)
+                .context(UnknownTransferSyntaxSnafu {
+                    ts_uid: transfer_syntax,
+                })?;
+        let ts_type = GDCMTransferSyntax::from_str(&registry.uid()).map_err(|_| {
+            UnsupportedTransferSyntaxSnafu {
+                ts: transfer_syntax.clone(),
+            }
+            .build()
+        })?;
+
+        let samples_per_pixel = samples_per_pixel(self).context(GetAttributeSnafu)?;
+        let bits_allocated = bits_allocated(self).context(GetAttributeSnafu)?;
+        let bits_stored = bits_stored(self).context(GetAttributeSnafu)?;
+        let high_bit = high_bit(self).context(GetAttributeSnafu)?;
+        let pixel_representation = pixel_representation(self).context(GetAttributeSnafu)?;
+        let planar_configuration = if let Ok(el) = self.element(tags::PLANAR_CONFIGURATION) {
+            let res = el.uint16();
+            res.unwrap_or(0)
+        } else {
+            0
+        };
+        let rescale_intercept = rescale_intercept(self);
+        let rescale_slope = rescale_slope(self);
+        let number_of_frames = number_of_frames(self).context(GetAttributeSnafu)?;
+        let voi_lut_function = voi_lut_function(self).context(GetAttributeSnafu)?;
+        let voi_lut_function = voi_lut_function.and_then(|v| VoiLutFunction::try_from(&*v).ok());
+
+        let decoded_pixel_data = match pixel_data.value() {
+            Value::PixelSequence(v) => {
+                let fragments = v.fragments();
+                let gdcm_error_mapper = |source: GDCMError| DecodeError::Custom {
+                    message: source.to_string(),
+                    source: Some(Box::new(source)),
+                };
+
+                let frame = frame as usize;
+                let data = if number_of_frames == 1 && fragments.len() > 1 {
+                    let mut buf = Vec::new();
+                    fragments
+                        .into_iter()
+                        .for_each(|fragment| buf.append(&mut fragment.clone()));
```

Comment:
I think a copy isn't needed if you use `extend` instead.

```suggestion
                        .for_each(|fragment| buf.extend(&fragment));
```

Reply:
This is a nice one, still I am trying to see if I can avoid the extra allocation, this one is so the first iteration works.